nimble_build_system/cad/rack_assembly.py

```diff
@@ -0,0 +1,471 @@
+"""
```

Comment:
I've had a skim through this. Clearly it works as we have the renders.

The real test will be how well we can get the information we need out for the instructions written. But I think I can only really confirm that when I start trying to generate them. If we need to do some tweaking it is probably worth doing it at that time.